tools/c7n_mailer/c7n_mailer/email_delivery.py

```diff
@@ -256,8 +256,10 @@ def send_c7n_email(self, sqs_message, email_to_addrs, mimetext_msg):
                     self.config,
                 )
             )
+            # TODO shall we raise the exception and interrupt the entire notify process?
```

Comment:
Hmm good question. I think since we call notifications out as best effort, logging the errors but not killing a policy run feels appropriate.

Reply:
Understood. I have a small concern regarding the case where only one notification mechanism is configured. In the event of a failure, user won't get any notification, and the SQS message will still be acknowledged (instead of remaining in the queue or being moved to a dead letter queue), which means there is no straightforward way to retransmit the notification.

Reply:
Hmm true, fair point. I suppose if we raise an exception here, we could handle it somewhere around [here](https://github.com/cloud-custodian/cloud-custodian/blob/c45b57994313184331e5f258eb1cc933275858f2/tools/c7n_mailer/c7n_mailer/sqs_queue_processor.py#L113-L116) and skip the `ack()` call without killing the policy run?